### PR TITLE
config(rest-api): enable authentication and api authorization by default

### PR DESCRIPTION
**Pull request by david-leifker**
* breaking change (previously disabled by default)
* upgraded pytests for running with authentication enabled

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

**Review by RyanHolstien — APPROVED** [review]

**Comment by Masterchen09** [comment]
@david-leifker First, I wanted to include this in my PR for the Helm chart...however I think here it is probably better...

I have already noticed it sometime ago - we have enabled the REST API authorizations for our deployment for some time - and I also have tried to fix it, however the whole policy logic is a bit hard to follow (i.e. the PolicyConfig.java is just too big and is confusing if you try to understand it...there are plain privileges, then they are grouped somehow in READ, WRITE, etc. and then there is CRUD-like structure which is used "somehow"): Currently some of the privileges which are affected by the REST_API_AUTHORIZATION_ENABLED environment variable cannot be assigned to custom policies using the UI because the privileges are missing in the corresponding list in [PoliciesConfig.java](https://github.com/datahub-project/datahub/blob/0187fc6c0fa1ce5a5e33982d483d4070540b2a43/metadata-utils/src/main/java/com/linkedin/metadata/authorization/PoliciesConfig.java#L176) (see also this comment [here](https://github.com/datahub-project/datahub/blob/0187fc6c0fa1ce5a5e33982d483d4070540b2a43/metadata-utils/src/main/java/com/linkedin/metadata/authorization/PoliciesConfig.java#L421-L496)). These privileges are only granted to the DataHub Root User and the Admin role, but cannot be included in custom policies.

Not sure, but I think this is something which must be checked if the REST API authorizations are enabled by default now, at least the privileges which have nothing to do with the GraphQL endpoints (as the comment linked above suggests, although I do not fully understand the logic why something should or should not be included in the list) should be available as privileges in the UI:

RESTORE_INDICES_PRIVILEGE
ES_EXPLAIN_QUERY_PRIVILEGE
GET_ES_TASK_STATUS_PRIVILEGE
GET_TIMESERIES_INDEX_SIZES_PRIVILEGE
SET_WRITEABLE_PRIVILEGE
APPLY_RETENTION_PRIVILEGE
PRODUCE_PLATFORM_EVENT_PRIVILEGE

**Comment by david-leifker** [comment]
@Masterchen09 - There is some history with how the policy privileges have evolved over time. Initially, APIs like OpenAPI and Rest.li were authenticated only with no authorization checks. This was added but there were privileges which were higher level concepts used by graphql which performs more complex actions (not simple CRUD operations).

The initial implementation of OpenAPI/Rest.li was build similar to graphql privileges where they were higher level and usually simple allow or deny. This simplicity didn't satisfy more complex access control rules which needed CRUD semantics for authorization. This was added to all APIs (where possible for graphql) in order to be able to have a somewhat consistent authorization policy across all APIs.

The privileges listed are privileges that were created for either OpenAPI or Rest.li and since these APIs were considered separate from graphql /UI they were never added to the UI to prevent confusion because there is no way to exercise them in the UI or via graphql.

While those privileges are not part of the UI, you can create policies via APIs to include them. As you pointed out the typical users for those privileges are admin or Admin role only and are not exactly useful for anyone but a user of the OpenAPI or Rest.li APIs (this does include the cli users).

Hopefully that explains how we got to this point. As to whether we should include these in the UI, even though they are not part of the UI experience, I'd want to get input from product @chriscollins3456 @jjoyce0510

**Comment by david-leifker** [comment]
I'd have to review some of those privileges, however it might be less confusing to add like `MANAGE_SYSTEM_OPERATIONS` to the UI and consolidate those other permissions into 1. The others would still be respected, but it would be less confusing to the average user.

**Comment by david-leifker** [comment]
See https://github.com/datahub-project/datahub/pull/11549
